src/components/segmented-control/index.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import {InteractiveColor} from '@atb/theme/colors';
+import {View, ViewStyle, StyleProp, TouchableOpacity} from 'react-native';
+import {StyleSheet, useTheme} from '@atb/theme';
+import ThemeText from '../text';
+
+export type SegmentOptions = {
+  onPress: () => void;
+  text: string;
+  subtext?: string;
+  selected?: boolean;
+  accessibilityHint?: string;
+};
+
+type SegmentedControlProps = {
+  color: InteractiveColor;
+  activeIndex: number;
+  options: SegmentOptions[];
+  enabled?: boolean;
+  style?: StyleProp<ViewStyle>;
+};
+
+export default function SegmentedControl({
+  color,
+  activeIndex,
+  options,
+  enabled = true,
+  style,
+}: SegmentedControlProps) {
+  const styles = useStyles();
+  const {theme} = useTheme();
+
+  const interactiveColor = theme.interactive[color];
+
+  return (
+    <View
+      style={[
+        styles.segmentedControl,
+        {
+          backgroundColor:
+            interactiveColor[enabled ? 'default' : 'disabled'].background,
+        },
+        style,
+      ]}
+    >
+      {options.map((option, i) => {
+        const selected = activeIndex === i;
+        const currentColor = interactiveColor[selected ? 'active' : 'default'];
+        const textColor = enabled
+          ? currentColor.text
+          : theme.text.colors.disabled;
```

Comment:
Will this work? My concern is that `theme.text.colors.disabled` will always be black in light mode (and vice versa), regardless of the background it is placed on?